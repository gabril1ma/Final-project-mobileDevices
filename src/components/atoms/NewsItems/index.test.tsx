import { create, ReactTestRendererJSON } from "react-test-renderer";
import NewsItem from "./newsItems";

export type TTreeRenderer = ReactTestRendererJSON | ReactTestRendererJSON[];

const MOCK_TEXT = "Jest test";
const mockFunction = () => jest.fn();

describe("<NewsItem />", () => {
  let tree: TTreeRenderer;
  let testTree: ReactTestRendererJSON;

  beforeEach(() => {
    tree = create(<NewsItem title={MOCK_TEXT} imageUrl={MOCK_TEXT} />).toJSON();

    testTree = tree[0] ?? tree;
  });

  it("should render component", () => {
    expect(testTree).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(testTree?.children?.length).toBe(2);
  });
});
