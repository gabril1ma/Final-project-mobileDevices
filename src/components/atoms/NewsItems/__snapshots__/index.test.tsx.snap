// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewsItem /> should render component 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderBottomColor": "#ccc",
      "borderBottomWidth": 1,
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <Image
    source={
      {
        "uri": "Jest test",
      }
    }
    style={
      {
        "borderRadius": 25,
        "height": 50,
        "marginRight": 10,
        "width": 50,
      }
    }
  />
  <Text
    style={
      {
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    Jest test
  </Text>
</View>
`;
